### DISCUSSION TITLE
Proxmox Virtual Environment 8

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1472] [Announcements]
It seems that Proxmox Virtual Environment 8 is on the horizon. With the recent update to `pve-manager: 7.4-13` , you can now utilize the command `pve7to8 --full` to check readiness for the transition.
# No longer in beta testing!
>If you are currently using the beta version, update normal, you will notice the availability of `pve-manager 8.0.3`. This update will remove the `test` repository but will not activate the `no-subscription` repository, so please ensure that you enable it. If you wish to include the `test` repository, you can add it by executing the "Proxmox VE Post Install" script.

**Comment by tteck** [comment]
The 8.x family is based on the great Debian 12 "Bookworm" and comes with a 6.2 kernel, QEMU 8.0.6, LXC 5.0.3, OpenZFS 2.1.12
https://forum.proxmox.com/threads/proxmox-ve-8-0-beta-released.128677/

**Comment by tteck** [comment]
First, utilize the command `pve7to8 --full` to check readiness for the transition.

This command is designed to transition a currently running PVE 7 instance into a running PVE 8 instance.
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/pve8-upgrade.sh)"
```
To ensure a proper source installation, please respond with "yes" to all questions prompted during the installation process.

**Reply by DSDV** [reply]
is this still working (since its not listed on the website). 
i would like to upgrade to stable pve8 (:

**Reply by tteck** [reply]
Yes, to transition a currently running PVE 7.4-13 or later instance into a running PVE 8.1.x instance.

**Comment by tteck** [comment]
Script was used...

![Screenshot 2023-06-10 10 26 56 PM](https://github.com/tteck/Proxmox/assets/21342670/74abd6f4-3541-4930-8c8c-be133aaff1fa)


![Screenshot 2023-06-10 9 28 12 PM](https://github.com/tteck/Proxmox/assets/21342670/973c3acf-cb5c-408e-b886-769d0fb772e2)

Run the "Proxmox VE Kernel Clean" script after the reboot to remove any unneeded  kernels.

**Reply by tteck** [reply]
I have not experienced any issues with PVE8 so far.

**Comment by oOStroudyOo** [comment]
I assume this means that I'd need to update the repo sources from bullseye to bookworm?

i.e, in your cloudflared script it sets `https://pkg.cloudflare.com/cloudflared bullseye main` as a source in `/etc/apt/sources.list.d/cloudflared.list` which now shows an "old suite 'bullseye' configured!" warning.

**Reply by tteck** [reply]
> I assume this means that I'd need to update the repo sources from bullseye to bookworm?

The `pve8-upgrade.sh` script facilitates the essential source modifications required for the installation of Proxmox VE 8.
 
Regarding cloudflared, if it is installed on the host machine that will be transformed into Debian 12 (bookworm): https://pkg.cloudflare.com/index.html#debian-bookworm (short answer, yes)

**Reply by tteck** [reply]
The Cloudflared LXC script in the repository creates its own container, it should operate independently from PVE8 and should not be impacted by any changes related to the Proxmox Virtual Environment version (bookworm).

**Comment by tteck** [comment]
The Proxmox VE Kernel Clean tool has been updated to be compatible with PVE8. When migrating from a running PVE7 instance to PVE8, the Kernel Clean tool can effectively eliminate any unnecessary 5.15.* kernels.

**Comment by danbayliss** [comment]
I've just upgraded and it seems to be working ok. Just a word of warning I didn't realise PVE8 was still a beta as i'm not 100% comfortable running a beta. As soon as PV8 is released i'm hoping it will be easy to update away from the beta releases

**Reply by tteck** [reply]
> As soon as PV8 is released i'm hoping it will be easy to update away from the beta releases

It will simply be a `apt` command.

**Comment by mitkovip** [comment]
Hi is it possible to update script "Proxmox VE Processor Microcode". I have installed beta directly but when i try to run this script i get: "Proxmox Virtual Environment Not Detected". It worked on PV7 without problems. Thanks

**Reply by tteck** [reply]
Should work with PVE8 now.

**Reply by mitkovip** [reply]
thanks for quick reply but i get folowing: 
✗ GenuineIntel
Intel(R) is not supported

**Reply by tteck** [reply]
fixed

**Reply by mitkovip** [reply]
Great, it is working now!
journalctl -k | grep -E "microcode: microcode"
Jun 20 22:13:46 pve kernel: microcode: microcode updated early to revision 0x2f, date = 2019-11-12

Thanks a lot for quick fix!

**Comment by danbayliss** [comment]
I'm seeing the following error in my logs once a day since I upgraded to PV8. Any ideas?
![vfXByvh](https://github.com/tteck/Proxmox/assets/54558493/f34869db-d21e-4d27-ae8f-9bc2ab49fe3e)
![z5uJBWt](https://github.com/tteck/Proxmox/assets/54558493/bfbce76c-de89-4127-b11e-08db716c559c)

If I manually run apt-get update I get the following

> Hit:1 http://ftp.debian.org/debian bookworm InRelease
> Hit:2 http://download.proxmox.com/debian/ceph-quincy bookworm InRelease                                    
> Get:3 http://security.debian.org/debian-security bookworm-security InRelease [48.0 kB]                                   
> Get:4 http://download.proxmox.com/debian/pve bookworm InRelease [1,818 B]                                                            
> Get:5 http://ftp.debian.org/debian bookworm-updates InRelease [52.1 kB]                                         
> Err:6 https://enterprise.proxmox.com/debian/pve bookworm InRelease                  
>   401  Unauthorized [IP: 51.91.38.34 443]
> Get:7 http://download.proxmox.com/debian/pve bookworm/pvetest amd64 Packages [70.9 kB]
> Reading package lists... Done       
> E: Failed to fetch https://enterprise.proxmox.com/debian/pve/dists/bookworm/InRelease  401  Unauthorized [IP: 51.91.38.34 443]
> E: The repository 'https://enterprise.proxmox.com/debian/pve bookworm InRelease' is not signed.
> N: Updating from such a repository can't be done securely, and is therefore disabled by default.
> N: See apt-secure(8) manpage for repository creation and user configuration details.
> W: Skipping acquire of configured file 'pve-no-subscription/binary-amd64/Packages' as repository 'http://download.proxmox.com/debian/pve bookworm InRelease' doesn't have the component 'pve-no-subscription' (component misspelt in sources.list?)
> W: Skipping acquire of configured file 'pve-no-subscription/i18n/Translation-en' as repository 'http://download.proxmox.com/debian/pve bookworm InRelease' doesn't have the component 'pve-no-subscription' (component misspelt in sources.list?)
> W: Skipping acquire of configured file 'pve-no-subscription/i18n/Translation-en_US' as repository 'http://download.proxmox.com/debian/pve bookworm InRelease' doesn't have the component 'pve-no-subscription' (component misspelt in sources.list?)

**Reply by tteck** [reply]
What repositories do you have enabled?

**Reply by danbayliss** [reply]
I don't remember changing anything in here so not sure if something was missed when it upgraded?
![zMg3nw9](https://github.com/tteck/Proxmox/assets/54558493/6127aa00-daa2-4f82-9416-5ce836083f26)

**Reply by tteck** [reply]
disable the enterprise repo

**Reply by danbayliss** [reply]
That's fixed it, thank you

**Comment by tteck** [comment]
Now out of beta! https://proxmox.com/en/news/press-releases/proxmox-virtual-environment-8-0

**Reply by David-Moreira** [reply]
Heyia @tteck 
I was trying to use your script to do an inplace upgrade.

Can you help with this?
![image](https://github.com/tteck/Proxmox/assets/22283161/f2dc4ab4-8bdd-432b-a0d0-7cbf4d54a678)

**Reply by tteck** [reply]
If you are currently using the beta version, update normal, you will notice the availability of `pve-manager 8.0.3`. This update will remove the `test` repository but will not activate the `no-subscription` repository, so please ensure that you enable it. If you wish to include the `test` repository, you can add it by executing the "Proxmox VE Post Install" script.

**Reply by tteck** [reply]
@David-Moreira I just switched repositories in the script, try again.

**Reply by David-Moreira** [reply]
Hello, it's going now, this time I pressed Yes on the pve test repository so I don't know what fixed it, but I'll assume it is the repository you switched?

I'll let you know if it completed successfuly. Thanks!

**Reply by David-Moreira** [reply]
In place update ran successfully with your script, everything seems to be working fine so far. Granted I don't have anything too fancy on my installation. 
I've also cleared old kernels with the Proxmox VE Kernel Clean tool.

Thanks for your work on these useful scripts. :)

**Reply by tteck** [reply]
Just curious, how many old kernels did it remove?

**Reply by David-Moreira** [reply]
It was 4. 
One was 5.* others were 6.*
Granted I have been using proxmox for a relatively short time, like a month or so.

**Reply by danbayliss** [reply]
All working and now off the beta branch. Thanks!

**Comment by GM-AZ** [comment]
just ran script on my dev server from 7 to 8. Worked perfectly. Only issue was after reboot the usb drive showed it was not active. After powering server off and on everything came back up. Ran and removed two 5. kernels.   
Thanks for the scripts

**Comment by bharvey88** [comment]
Writing this out since it's split between multiple messages:

run `apt-get update` and `apt-get dist-upgrade` first ->then that allows you to run `pve7to8 --full` -> then run his script to actually update it  
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/pve8-upgrade.sh)"
```
 -> then reboot -> then run Proxmox VE Kernel Clean -> then if needed run the microcode script if needed.

Is this accurate?

**Reply by tteck** [reply]
Cleaned it up a bit... 😉

**Reply by lexybarton** [reply]
for me, apt-update, dist-upgrade are not working, but **`apt-get update `** and **`apt-get dist-upgrade`** , then `pve7to8 --full` ...

**Comment by lyricnz** [comment]
Yup.  Applied to two nodes, worked fine.

**Comment by saxon-viking** [comment]
Worked fine for me first go, thanks tteck for all your hard work

**Comment by SrLPZ** [comment]
![Screenshot_2023-06-29-15-57-25-54_3d419158bad5872c40592a6c9956e692~2](https://github.com/tteck/Proxmox/assets/76914115/455b9477-3e0b-47b7-addf-7d81d7f7c26f)
I get the following warning when I run the check with the `pve7to8 --full` command.  

Does anyone know what it is and how to fix it?

**Reply by tteck** [reply]
It is letting you know that 5 instances are running, whether it's containers or virtual machines. It is safe to ignore that warning.

**Comment by MBj1703** [comment]


**Reply by tteck** [reply]
Show the sources from the following commands (one at a time)

```
cat /etc/apt/sources.list
cat /etc/apt/sources.list.d/pve-enterprise.list
cat /etc/apt/sources.list.d/pve-install-repo.list
cat /etc/apt/sources.list.d/ceph.list
cat /etc/apt/sources.list.d/pvetest-for-beta.list
```

**Reply by MBj1703** [reply]
I fixed it, thank you

**Reply by tteck** [reply]
What was your solution?

**Reply by MBj1703** [reply]
missing source for Ceph, even not using t nor installed it

**Comment by BliXem1** [comment]
Hi, Can I use the script on Proxmox 8+ to remove the annoying message with login and replace the repo's?

**Reply by tteck** [reply]
Use the "Proxmox VE Post Install" script

**Reply by DJKatastrof** [reply]
Hey!
I just upgraded to 8 using your script, but using Post Install script, the subscription message won't disappear. 
Any ideas?

**Reply by tteck** [reply]
Clear browser cache.

**Reply by migsun** [reply]
With 8.1.4 the nag screen appears again and running the Post Script again and clearing the browser cache could not help.

**Reply by tteck** [reply]
If the no-nag was installed, running the Post install script again will not display that option.
Here are some checks:
1) `grep "NoMoreNagging" /usr/share/javascript/proxmox-widget-toolkit/proxmoxlib.js`  if the output is `.data.status.toLowerCase() == 'NoMoreNagging') {` it's installed.
2) `cat /etc/apt/apt.conf.d/no-nag-script` shows the actual script.
3) `apt --reinstall install proxmox-widget-toolkit`

```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 0 not upgraded.
Need to get 210 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://download.proxmox.com/debian/pve bookworm/pve-no-subscription amd64 proxmox-widget-toolkit all 4.1.3 [210 kB]
Fetched 210 kB in 1s (154 kB/s)                
(Reading database ... 53249 files and directories currently installed.)
Preparing to unpack .../proxmox-widget-toolkit_4.1.3_all.deb ...
Unpacking proxmox-widget-toolkit (4.1.3) over (4.1.3) ...
Setting up proxmox-widget-toolkit (4.1.3) ...
Removing subscription nag from UI...
```
Displays "**Removing subscription nag from UI...**"

**Reply by DSDV** [reply]
@migsun i run a 8.1.4 with an up to date system as of now and the post install script works as desired on my systems. 
if ttecks troubleshooting tips don't help maybe check with a fresh install on a spare system?

**Reply by migsun** [reply]
After reinstalling the widget-toolkit, post-install script and restarting the server, the nag screen stopped appearing. Thank you very much!
I think I had problems with my first attempt to manually hide the subscription message in the proxmoxlib file.
